pkg/security/probe/probe_ebpf.go

```
@@ -1678,43 +1546,182 @@ func (p *EBPFProbe) handleEvent(CPU int, data []byte) {
 		}
 		event.Setrlimit.Target = &pce.ProcessContext
 	case model.CapabilitiesEventType:
-		if _, err = event.CapabilitiesUsage.UnmarshalBinary(data[offset:]); err != nil {
-			seclog.Errorf("failed to decode capabilities usage event: %s (offset %d, len %d)", err, offset, len(data))
-			return
+		if err = p.regularUnmarshalEvent(&event.CapabilitiesUsage, eventType, offset, dataLen, data); err != nil {
+			return false
 		}
 		if event.CapabilitiesUsage.Attempted == 0 && event.CapabilitiesUsage.Used == 0 {
 			seclog.Debugf("capabilities usage event with no attempted or used capabilities, skipping")
-			return
+			return false
 		}
 		// is this thread-safe?
 		event.ProcessCacheEntry.CapsAttempted |= event.CapabilitiesUsage.Attempted
 		event.ProcessCacheEntry.CapsUsed |= event.CapabilitiesUsage.Used
 	case model.PrCtlEventType:
-		if _, err = event.PrCtl.UnmarshalBinary(data[offset:]); err != nil {
-			seclog.Errorf("failed to decode prctl event: %s (offset %d, len %d)", err, offset, len(data))
-			return
+		if err = p.regularUnmarshalEvent(&event.PrCtl, eventType, offset, dataLen, data); err != nil {
+			return false
 		}
 		if event.PrCtl.IsNameTruncated {
 			p.MetricNameTruncated.Add(1)
 		}
 	}
+	return true
+}
 
-	// send related events
-	for _, relatedEvent := range relatedEvents {
-		p.DispatchEvent(relatedEvent, true)
-		p.eventPool.Put(relatedEvent)
+func (p *EBPFProbe) handleBeforeProcessContext(eventType model.EventType, event *model.Event, data []byte, offset int, dataLen uint64, newEntryCb func(entry *model.ProcessCacheEntry, err error)) bool {
```

same here, could you return an error instead of a bool plz ?

---

yes, same as the other one